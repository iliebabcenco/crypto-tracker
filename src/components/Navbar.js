import SearchForm from '../containers/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
