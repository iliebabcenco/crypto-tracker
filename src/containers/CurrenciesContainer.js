import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import fetchCurrencies from '../api/fetcher';
import Currency from '../components/Currency';

const CurrenciesContainer = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCurrencies());
  }, []);
  let currencies = useSelector((state) => state.items);
  const filterPattern = useSelector((state) => state.pattern);
  if (filterPattern) {
    currencies = currencies.filter((c) => c.name.toLowerCase().startsWith(filterPattern)
      || c.symbol.toLowerCase().startsWith(filterPattern));
  }
  return (

    <div className="currencies-container">
      <h3>Name</h3>
      <p>Description</p>
      {currencies ? currencies.map((currency) => (
        <Currency key={currency.name} currency={currency} />
      )) : 'Please wait'}
    </div>
  );
};

export default CurrenciesContainer;
