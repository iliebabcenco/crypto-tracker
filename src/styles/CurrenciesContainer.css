.currencies-container {
  margin: 4rem auto 4rem auto;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 70%;
  color: white;
  border: 1px solid #c9ab81;
  border-radius: 10px;
  background-color: rgba(62, 87, 104, 0.8);
  padding: 2rem;
}

.top-btn {
  border: none;
  background: none;
  color: white;
  font-size: 1.5rem;
}

.top-btn:hover {
  color: #c9ab81;
}

.top-container-div {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

@media only screen and (max-width: 768px) {
  .currencies-container {
    width: 98%;
  }

  .top-btn {
    font-size: 1rem;
  }

  .top-container-div h3 {
    font-size: 1rem;
  }
}
