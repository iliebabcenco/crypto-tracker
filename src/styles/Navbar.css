.navbar {
  width: 100%;
  color: white;
  display: flex;
  margin: 0 auto;
  padding-bottom: 1rem;
  padding-top: 1rem;
  justify-content: space-around;
  border-bottom: 1px solid #c9ab81;
  background-color: rgba(21, 30, 36, 0.8);
}