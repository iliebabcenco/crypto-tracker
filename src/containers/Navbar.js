import SearchForm from '../components/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
