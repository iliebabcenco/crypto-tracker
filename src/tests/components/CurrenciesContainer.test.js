import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import CurrenciesContainer from '../../containers/CurrenciesContainer';
import store from '../../store/store';
import Navbar from '../../components/Navbar';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><CurrenciesContainer /></Provider>);
  expect(asFragment(<Provider store={store}><CurrenciesContainer /></Provider>)).toMatchSnapshot();
});

it('displays correct first child', () => {
  const { container } = render(<Provider store={store}><CurrenciesContainer /></Provider>);
  expect(container.firstChild).toMatchSnapshot();
});

it('negative check', () => {
  render(<Provider store={store}><Navbar /></Provider>);
  expect(screen.queryByText('Letâ€™s do some math!')).not.toBeInTheDocument();
});
