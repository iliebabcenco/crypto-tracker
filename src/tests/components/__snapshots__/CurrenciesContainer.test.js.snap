// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct first child 1`] = `
<div
  class="currencies-container"
>
  <div
    class="top-container-div"
  >
    <h3>
      Top
       
      20
       
      cryptocurrencies:
    </h3>
    <button
      class="top-btn"
      type="button"
    >
      <i
        class="fas fa-arrow-right"
      />
       
      Show top
       
      100
    </button>
  </div>
</div>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="currencies-container"
  >
    <div
      class="top-container-div"
    >
      <h3>
        Top 20 cryptocurrencies:
      </h3>
      <button
        class="top-btn"
        type="button"
      >
        <i
          class="fas fa-arrow-right"
        />
         Show top 100
      </button>
    </div>
  </div>
</DocumentFragment>
`;
